swap/swap_test.go

```diff
@@ -1083,3 +1084,29 @@ func (d *dummyMsgRW) ReadMsg() (p2p.Msg, error) {
 func (d *dummyMsgRW) WriteMsg(msg p2p.Msg) error {
 	return nil
 }
+
+func TestFileLogger(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
+	handler, err := log.FileHandler("/home/git/go/src/github.com/ethersphere/swarm/testlog/test.log", log.TerminalFormat(true))
+	if err != nil {
+		t.Fatal(err)
+	}
+	log.Root().SetHandler(log.LvlFilterHandler(log.Lvl(*loglevel), handler))
+	log.Error("Test")
+}
+
+func TestRotatingFileLogger(t *testing.T) {
```

Comment:
These unit tests (on both file loggers) seem to test only the working of the logger itself. I assume this is already tested thoroughly by the authors of the `log` package and hence don't see the purpose of replicating this in this context.

Furthermore, running these test gives me a `permission denied` error.

Reply:
I was not sure about the actual implementation of the rotating file handler.

I wanted to test the file handler before integrating it into the swap package.

This test is a mock of the actual test I have to do.

I think the path is hardcoded in this test.
That's why you get a permission denied.

I will remove this test and generate something more appropriate for this implementation.

Reply:
Right, there is no need to test the actual behavior of other packages